src/components/regularRotationList/RestaurantList.tsx

```diff
@@ -1,160 +1,59 @@
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">Mediterranean</Typography>
-          </div>
-          {mediterraneanRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">American</Typography>
-          </div>
-          {americanRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">Mexican</Typography>
-          </div>
-          {mexicanRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">Italian</Typography>
-          </div>
-          {italianRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">Brunch/Breakfast/Snack</Typography>
-          </div>
-          {brunchRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
-          ))}
-          <div className={classes.genre}>
-            <Typography variant="h3">Boba</Typography>
-          </div>
-          {bobaRestaurants.map((restaurant) => (
-            <div className={classes.name}>
-              <Link to={restaurant} key={restaurant} className={classes.links}>
-                <Typography>{restaurant}</Typography>
-              </Link>
-            </div>
+          {restaurantTypes.map((genre) => (
+            <Fragment key={genre}>
+              <div className={classes.genre}>
+                <Typography variant="h3">{genre}</Typography>
+              </div>
+              {Array.from(RestaurantMap)
```

Comment:
Is this weird to do? It feels weird to me (converting a map into an array just to filter through it and map out links)

Reply:
I've realized part of my problem is not sticking to a functional approach of returning a new object and instead modifying Map. Which if i had a true backend (which I do not) would not be an issue because i would be fetching this from the backend.